### PR TITLE
Spring 1

### PR DESCRIPTION
**Pull request by RuslanIbIsh**

### CODE REVIEW

**Review by Ilya-brinchuk — APPROVED** [review]

File: src/main/java/com/iri/spring/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Repository;
+
+@Primary
```
Comment (Sofasmile) [inline src/main/java/com/iri/spring/dao/impl/UserDaoImpl.java:14]: do we need this annotation here

File: src/main/java/com/iri/spring/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+@Repository
+public class UserDaoImpl implements UserDao {
+    @Autowired
+    private SessionFactory sessionFactory;
```
Comment (Sofasmile) [inline src/main/java/com/iri/spring/dao/impl/UserDaoImpl.java:18]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_01.html
> What is better: setter, constructor, or field injection? Think why.

File: src/main/java/com/iri/spring/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private UserDao userDao;
+
+    @Autowired
+    public UserServiceImpl(@Qualifier(value = "userDaoImpl") UserDao userDao) {
```
Comment (Sofasmile) [inline src/main/java/com/iri/spring/service/impl/UserServiceImpl.java:16]: do we need Qualifier here?

File: src/main/java/com/iri/spring/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import com.iri.spring.model.User;
+import com.iri.spring.service.UserService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private UserDao userDao;
```
Comment (Sofasmile) [inline src/main/java/com/iri/spring/service/impl/UserServiceImpl.java:13]:
```suggestion
    private final UserDao userDao;
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by AntonPanchishko — APPROVED** [review]

File: src/main/java/com/iri/spring/dao/impl/UserDaoImpl.java (hunk 1 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/spring/dao/impl/UserDaoImpl.java:14]: just for remember

File: src/main/java/com/iri/spring/dao/impl/UserDaoImpl.java (hunk 2 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/spring/dao/impl/UserDaoImpl.java:18]: throw Constructor app does not run if smth
throw the field = NPE

File: src/main/java/com/iri/spring/service/impl/UserServiceImpl.java (hunk 3 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/spring/service/impl/UserServiceImpl.java:16]: Just for remember

File: src/main/java/com/iri/spring/service/impl/UserServiceImpl.java (hunk 4 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/spring/service/impl/UserServiceImpl.java:13]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**